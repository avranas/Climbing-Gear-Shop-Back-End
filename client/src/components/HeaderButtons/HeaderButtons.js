import { Link, useNavigate } from "react-router-dom";
import { useState, useEffect } from "react";
import axios from "axios";

const HeaderButtons = (props) => {
  const navigate = useNavigate();
  const [userIsLoggedIn, setUserIsLoggedIn] = useState(null);

  const handleLogout = async () => {
    navigate("/logout");
  };

  useEffect(() => {
    const checkAuthentication = async () => {
      try {
        const response = await axios("/authenticated");
        setUserIsLoggedIn(response.data);
      } catch (err) {
        console.log(err);
      }
    };
    checkAuthentication();
  }, []);

  return (
    <div id="header-user-buttons">
      {userIsLoggedIn === null || userIsLoggedIn === false ? (
        <div>
          <Link to="/login">
            <button id="left-button" className="small-button">
              Login
            </button>
          </Link>
          <Link to="/register">
            <button type="button" className="small-button">
              Register
            </button>
          </Link>
        </div>
      ) : (
        <div>
          <button onClick={handleLogout} className="small-button">
            Logout
          </button>

          <Link to="/orders/0">
            <button type="button" className="small-button">
              Orders
            </button>
          </Link>
        </div>
      )}
    </div>
  );
};

export default HeaderButtons;
