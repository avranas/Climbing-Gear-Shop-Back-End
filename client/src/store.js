import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});