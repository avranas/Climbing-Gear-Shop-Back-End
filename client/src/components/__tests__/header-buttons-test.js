import { cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import HeaderButtons from "../HeaderButtons/HeaderButtons";

afterEach(() => {
  cleanup();
});

const headerButtons = (
  <Provider store={store}>
    <BrowserRouter>
      <HeaderButtons />
    </BrowserRouter>
  </Provider>
);

test("HeaderUserButtons matches snapshot", async () => {
  const tree = renderer.create(headerButtons).toJSON();
  expect(tree).toMatchSnapshot();
});
 