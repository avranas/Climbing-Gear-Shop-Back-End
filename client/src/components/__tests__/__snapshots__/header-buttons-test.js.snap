// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUserButtons matches snapshot 1`] = `
<div
  id="header-user-buttons"
>
  <div />
</div>
`;
